nixos/modules/services/networking/inadyn.nix

```diff
@@ -0,0 +1,199 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.inadyn;
+
+  # check if a value of an attrset is not null or an empty collection
+  nonEmptyValue = _: v: ! isNull v && v != [ ] && v != { };
+
+  renderOption = k: v:
+    if builtins.elem k [ "provider" "custom" ] then
+      lib.concatStringsSep "\n"
+        (mapAttrsToList
+          (name: config: ''
+            ${k} ${name} {
+                ${lib.concatStringsSep "\n    " (mapAttrsToList renderOption (filterAttrs nonEmptyValue config))}
+            }'')
+          v)
+    else if k == "include" then
+      "${k}(\"${v}\")"
+    else if k == "hostname" && builtins.isList v then
+      "${k} = { ${builtins.concatStringsSep ", " (map (s: "\"${s}\"") v)} }"
+    else if builtins.isBool v then
+      "${k} = ${boolToString v}"
+    else if builtins.isString v then
+      "${k} = \"${v}\""
+    else
+      "${k} = ${toString v}";
+
+  configFile' = pkgs.writeText "inadyn.conf"
+    ''
+    # This file was generated by nix
+    # do not edit
+
+    ${(lib.concatStringsSep "\n" (mapAttrsToList renderOption (filterAttrs nonEmptyValue cfg.settings)))}
+    '';
+
+  configFile = if (cfg.configFile != null) then cfg.configFile else configFile';
+in
+{
+  options.services.inadyn = with types;
+    let providerOptions =
+      {
+        include = mkOption {
+          default = null;
+          description = mdDoc "File to include additional settings for this provider from.";
+          type = nullOr path;
+        };
+        ssl = mkOption {
+          default = true;
+          description = mdDoc "Whether to use HTTPS for this DDNS provider.";
+          type = bool;
+        };
+        username = mkOption {
+          default = null;
+          description = mdDoc "Username for this DDNS provider.";
+          type = nullOr str;
+        };
+        password = mkOption {
+          default = null;
+          description = mdDoc ''
+            Password for this DDNS provider.
+
+            WARNING: This will be world-readable in the nix store.
+            To store credentials securely, use the `include` or `configFile` options.
+          '';
+          type = nullOr str;
+        };
+        hostname = mkOption {
+          default = "*";
+          example = "your.cool-domain.com";
+          description = mdDoc "Hostname alias(es).";
+          type = either str (listOf str);
+        };
+      };
+    in
+    {
+      enable = mkEnableOption (mdDoc ''
+        synchronise your machine's IP address with a dynamic DNS provider using inadyn.
```

Comment:
mkEnableOption adds a dot at the end, so please remove the extra dot at the end here (above).

Reply:
Should I open a new PR or can I join this one? I removed a great deal of options and simplified the config file rendering. OP had some neat tricks to accomodate the somewhat nested config structure, but I'm not sure it's necessary for such a simple module when include is available.
https://github.com/ahbk/my-nixos/blob/master/inadyn/default.nix